File: cleanup_corrupted_files.py
#!/usr/bin/env python3
import os
import re
import glob

def is_corrupted_file(file_path):
    """Check if a file is corrupted"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Check for common corruption patterns
        corruption_patterns = [
            r'^[a-z-]+$',  # Lines with only lowercase letters and hyphens
            r'^cursor/',   # Lines starting with cursor/
            r'^website-audit',  # Lines starting with website-audit
            r'^fix-errors',     # Lines starting with fix-errors
            r'^\s*[a-z-]+\s*$',  # Lines with only lowercase words
        ]
        
        lines = content.split('\n')
        for line in lines[:10]:  # Check first 10 lines
            for pattern in corruption_patterns:
                if re.match(pattern, line.strip()):
                    return True
        
        return False
    except:
        return True

def create_clean_placeholder(file_path):
    """Create a clean placeholder file"""
    try:
        # Extract the service name from the path
        path_parts = file_path.split('/')
        service_name = path_parts[-2] if path_parts[-1] == 'page.tsx' else path_parts[-1].replace('.tsx', '')
        
        # Convert to PascalCase
        service_name_pascal = ''.join(word.capitalize() for word in service_name.replace('-', ' ').split())
        
        clean_content = f'''import React from 'react'
import {{ Helmet }} from 'react-helmet-async'
import {{ Link }} from 'react-router-dom'
import {{ ArrowRight }} from 'lucide-react'

export default function {service_name_pascal}Page() {{
  return (
    <div className="min-h-screen bg-gradient-to-br from-slate-900 via-purple-900 to-slate-900 pt-20">
      <Helmet>
        <title>{service_name.replace('-', ' ').title()} - Zion Tech Group</title>
        <meta name="description" content="Professional {service_name.replace('-', ' ')} services by Zion Tech Group. Transform your business with our expert solutions." />
      </Helmet>
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-16 text-center">
        <h1 className="text-4xl font-bold text-white mb-6">{service_name.replace('-', ' ').title()}</h1>
        <p className="text-lg text-gray-300 mb-8">Professional {service_name.replace('-', ' ')} services coming soon.</p>
        <Link
          to="/contact"
          className="bg-gradient-to-r from-cyan-500 to-purple-600 text-white px-8 py-4 rounded-lg font-semibold hover:from-cyan-600 hover:to-purple-700 transition-all duration-300 flex items-center justify-center mx-auto w-fit"
        >
          Contact Us
          <ArrowRight className="w-5 h-5 ml-2" />
        </Link>
      </div>
    </div>
  )
}}'''
        
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(clean_content)
        
        print(f"Created clean placeholder: {file_path}")
        return True
    except Exception as e:
        print(f"Error creating placeholder for {file_path}: {e}")
        return False

def main():
    # Find all page files
    page_files = glob.glob('app/**/page.tsx', recursive=True)
    
    fixed_count = 0
    for file_path in page_files:
        if is_corrupted_file(file_path):
            if create_clean_placeholder(file_path):
                fixed_count += 1
    
    print(f"Fixed {fixed_count} corrupted files")

if __name__ == "__main__":
    main()

File: cleanup-corrupted-files.js
const fs = require('fs');
const path = require('path');

// List of known corrupted file patterns to remove
const corruptedPatterns = [
  'page-backup.tsx',
  'page-fixed.tsx',
  'page-broken',
  '.tsx.bak',
  '.ts.bak',
  '.js.bak'
];

// List of directories to clean up
const directoriesToClean = [
  'app/components',
  'app/hooks',
  'app/utils',
  'app/types',
  'app/pages',
  'app/config',
  'src',
  'components'
];

// Function to remove corrupted files
function removeCorruptedFiles(dir) {
  let removedCount = 0;
  
  try {
    const files = fs.readdirSync(dir);
    
    for (const file of files) {
      const filePath = path.join(dir, file);
      const stat = fs.statSync(filePath);
      
      if (stat.isDirectory()) {
        // Recursively clean subdirectories
        removedCount += removeCorruptedFiles(filePath);
      } else {
        // Check if file matches corrupted patterns
        const shouldRemove = corruptedPatterns.some(pattern => 
          file.includes(pattern) || file.endsWith(pattern)
        );
        
        if (shouldRemove) {
          fs.unlinkSync(filePath);
          console.log(`Removed corrupted file: ${filePath}`);
          removedCount++;
        }
      }
    }
  } catch (error) {
    console.error(`Error processing directory ${dir}:`, error.message);
  }
  
  return removedCount;
}

// Function to clean up specific problematic files
function cleanSpecificFiles() {
  const filesToClean = [
    'app/404.tsx',
    'app/force-dynamic.ts',
    'app/main.tsx',
    'app/root-layout.tsx',
    'app/service-template.tsx',
    'app/data/servicesData.ts',
    'app/micro-saas-services/microSaasServices.tsx',
    'app/micro-saas-services/services.tsx'
  ];
  
  let cleanedCount = 0;
  
  filesToClean.forEach(filePath => {
    try {
      if (fs.existsSync(filePath)) {
        fs.unlinkSync(filePath);
        console.log(`Removed problematic file: ${filePath}`);
        cleanedCount++;
      }
    } catch (error) {
      console.error(`Error removing ${filePath}:`, error.message);
    }
  });
  
  return cleanedCount;
}

// Function to create a simple 404 page
function createSimple404() {
  const content = `export default function NotFound() {
  return (
    <div className="min-h-screen bg-gradient-to-br from-slate-900 via-purple-900 to-slate-900 flex items-center justify-center">
      <div className="text-center">
        <h1 className="text-6xl font-bold text-white mb-4">404</h1>
        <p className="text-xl text-gray-300 mb-8">Page Not Found</p>
        <a href="/" className="bg-cyan-500 text-white px-6 py-3 rounded-lg hover:bg-cyan-600 transition-colors">
          Go Home
        </a>
      </div>
    </div>
  );
}`;
  
  try {
    fs.writeFileSync('app/not-found.tsx', content);
    console.log('Created simple 404 page');
  } catch (error) {
    console.error('Error creating 404 page:', error.message);
  }
}

// Main cleanup function
function cleanup() {
  console.log('Starting cleanup of corrupted files...');
  
  let totalRemoved = 0;
  
  // Remove corrupted files from all directories
  directoriesToClean.forEach(dir => {
    if (fs.existsSync(dir)) {
      console.log(`Cleaning directory: ${dir}`);
      totalRemoved += removeCorruptedFiles(dir);
    }
  });
  
  // Clean specific problematic files
  console.log('Cleaning specific problematic files...');
  totalRemoved += cleanSpecificFiles();
  
  // Create simple 404 page
  createSimple404();
  
  console.log(`Cleanup complete. Removed ${totalRemoved} files.`);
}

// Run cleanup
cleanup();